### DISCUSSION TITLE
[Idea] Backoffice product page: Display favorite modules icons along with tabs

### DISCUSSION THREAD

**Discussion by unlocomqx** [PrestaShop/PrestaShop#34990] [Ideas]
### The problem

Context: Product Admin page
Opening a module configuration is always 2 clicks away plus looking through the modules list and sometimes scrolling to find the module

![CleanShot 2024-01-07 at 18 48 52](https://github.com/PrestaShop/PrestaShop/assets/793712/498b467e-80c9-47f2-a5b2-965548ff20a5)

### The solution

Add a module to favorites for easier access. This would display the module icon along with tabs

![CleanShot 2024-01-07 at 18 49 43](https://github.com/PrestaShop/PrestaShop/assets/793712/2597aa29-4a12-41f6-9f17-48ec95d9a1ac)

I implemented this for my modules but it's better if it's a PrestaShop feature

Suggested UI: Add to favorite / Pin module tab
![CleanShot 2024-01-07 at 18 52 11](https://github.com/PrestaShop/PrestaShop/assets/793712/fad1e393-064d-4664-b201-2f08115d268a)

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by MatShir** [comment]
Hi @unlocomqx,
I'm not entirely sure about the requirements here. Could you provide more information about your use case?

The default setup looks fine to me. However, after installing the two modules in my shop, I noticed the addition of three new tabs: the tab module, tab product designer, and the tab dynamic product. Upon review, it seems that the tab module may be redundant and could potentially be removed. What are your thoughts on this?

**Reply by MatShir** [reply]
I'm talking with the new product page

**Reply by unlocomqx** [reply]
The "Modules" tab can be preserved as it would display the modules list as usual. But additional tabs can be added displaying favorited/pinned modules.
So the admin can pin any module of their choice for easy and quick access.

**Reply by MatShir** [reply]
If I understand correctly, you want to hide the unnecessary module tabs that are displayed by default.

**Reply by unlocomqx** [reply]
I don't want to hide any existing elements :)) I just want to pin certain modules and display them as a separate tab for easy access. So instead of opening the modules tab then clicking the module "Configure" button, you will access the module settings with one click on the pinned tab like this for example

![image](https://github.com/PrestaShop/PrestaShop/assets/793712/7e1273a3-cd20-4dda-8c7f-21706dc42b82)

**Reply by MatShir** [reply]
Haha sorry, maybe it's late or I need some rest from the latest Xmas & New Year night. But the tabs are already available on the page: product designer and product dynamic when the module is installed. You can switch on the module to configure, but they are all displayed so I don't see why we would need to pin them.
I'll try again tomorrow morning ^^

**Reply by unlocomqx** [reply]
😄 We'll discuss this tomorrow. It was a long day :))

**Reply by kpodemski** [reply]
hey @unlocomqx :)

You can add extra tabs on the new product page, which is the only supported page nowadays. Still, I understand that you want to have some quick access only for favorite modules, even those that still need to implement a custom tab inside the new product page. I must admit it's an interesting idea. Maybe we could think of a better way of placing it inside the UI so it is still visible somewhere directly, but without creating "a tab" where the space is limited anyway.

**Reply by unlocomqx** [reply]
Okay, I will see if there's a better way to show module icons in the product edit page. If you have any idea, please share

Thank you
